pkg/controller/volume/attachdetach/testing/testvolumespec.go

```diff
@@ -174,15 +174,20 @@ func CreateTestClient() *fake.Clientset {
 		return true, obj, nil
 	})
 	volumeAttachments = &storagev1.VolumeAttachmentList{}
+	var mu sync.Mutex
 	fakeClient.AddReactor("list", "volumeattachments", func(action core.Action) (handled bool, ret runtime.Object, err error) {
 		obj := &storagev1.VolumeAttachmentList{}
+		mu.Lock()
 		obj.Items = append(obj.Items, volumeAttachments.Items...)
+		mu.Unlock()
 		return true, obj, nil
 	})
 	fakeClient.AddReactor("create", "volumeattachments", func(action core.Action) (handled bool, ret runtime.Object, err error) {
 		createAction := action.(core.CreateAction)
 		va := createAction.GetObject().(*storagev1.VolumeAttachment)
+		mu.Lock()
 		volumeAttachments.Items = append(volumeAttachments.Items, *va)
+		mu.Unlock()
 		return true, createAction.GetObject(), nil
 	})
 
```

Comment:
don't we need the same locks for the pvs list/create handlers?

Reply:
This data race is concurrent read and write in `volumeAttachments.Items`, but i did not reproduce it in local env, so just add lock for attach case.

Reply:
by inspection, both need locks, so go ahead and add those, then lgtm